sycl/source/detail/queue_impl.cpp

```diff
@@ -400,6 +400,63 @@ pi_native_handle queue_impl::getNative() const {
   return Handle;
 }
 
+bool queue_impl::ext_oneapi_empty() const {
+  // If we have in-order queue where events are not discarded then just check
+  // the status of the last event.
+  if (isInOrder() && !MDiscardEvents) {
```

Comment:
we can do similar in L0 Plugin even for discarded-events, right?

Reply:
Yes, and we support it for discarded events. This if branch is just handling the case when events are not discarded by using the last command event.

Reply:
I don't see how you handle in-order queue in L0 plugin with the latest changes

Reply:
Indeed, in-order queue was handled in SYCL RT level only.
Added in-order queue handling to L0 plugin as well.